tests/unit/python/foglamp/plugins/south/coap_listen/test_coap_listen.py

```diff
@@ -0,0 +1,244 @@
+    config['uri']['value'] = config['uri']['default']
+
+    log_exception = mocker.patch.object(coap_listen._LOGGER, "exception")
+    log_info = mocker.patch.object(coap_listen._LOGGER, "info")
+    assert coap_listen.aiocoap_ctx is None
+
+    # WHEN
+    loop = asyncio.get_event_loop()
+    loop.call_soon(coap_listen.plugin_start(config))
+
+    # THEN
+    assert coap_listen.aiocoap_ctx is not None
+    assert 1 == log_info.call_count
+    calls = [
+        call('CoAP listener started on port {} with uri {}'.format(config['port']['value'], config['uri']['value']))]
+    log_info.assert_has_calls(calls, any_order=True)
+
+
+def test_plugin_reconfigure(mocker, unused_port):
+    # GIVEN
+    config = copy.deepcopy(_DEFAULT_CONFIG(unused_port()))
+    config['port']['value'] = config['port']['default']
+    config['uri']['value'] = config['uri']['default']
+    mocker.patch.object(coap_listen, '_plugin_stop', return_value=True)
+    log_exception = mocker.patch.object(coap_listen._LOGGER, "exception")
+    log_info = mocker.patch.object(coap_listen._LOGGER, "info")
+
+    # WHEN
+    new_config = coap_listen.plugin_reconfigure(config, _NEW_CONFIG)
+
+    # THEN
+    assert _NEW_CONFIG == new_config
+    assert 2 == log_info.call_count
+    # TODO: log_info.assert_called_with('Restarting HTTP_SOUTH plugin due to change in configuration keys [uri, port, host]')
+
+
+def test_plugin__stop(mocker, unused_port):
+    # GIVEN
+    config = copy.deepcopy(_DEFAULT_CONFIG(unused_port()))
+    config['port']['value'] = config['port']['default']
+    config['uri']['value'] = config['uri']['default']
+    log_exception = mocker.patch.object(coap_listen._LOGGER, "exception")
+    log_info = mocker.patch.object(coap_listen._LOGGER, "info")
+
+    # WHEN
+    loop = asyncio.get_event_loop()
+    loop.call_soon(coap_listen.plugin_start(config))
+    coap_listen._plugin_stop(config)
+
+    # THEN
+    assert 2 == log_info.call_count
+    calls = [call('CoAP listener started on port {} with uri sensor-values'.format(config['port']['value'])),
+             call('Stopping South COAP plugin...')]
+    log_info.assert_has_calls(calls, any_order=True)
+    assert 0 == log_exception.call_count
+
+
+def test_plugin_shutdown(mocker, unused_port):
+    # GIVEN
+    config = copy.deepcopy(_DEFAULT_CONFIG(unused_port()))
+    config['port']['value'] = config['port']['default']
+    config['uri']['value'] = config['uri']['default']
+    log_exception = mocker.patch.object(coap_listen._LOGGER, "exception")
+    log_info = mocker.patch.object(coap_listen._LOGGER, "info")
+
+    # WHEN
+    loop = asyncio.get_event_loop()
+    loop.call_soon(coap_listen.plugin_start(config))
+    coap_listen.plugin_shutdown(config)
+
+    # THEN
+    assert 3 == log_info.call_count
+    calls = [call('CoAP listener started on port {} with uri sensor-values'.format(config['port']['value'])),
+             call('Stopping South COAP plugin...'),
```

Comment:
In which part of the code are these calls made to logger for shutdown?

Reply:
first call in plugin_start and second call in _plugin_stop.